plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/ProductPicker.tsx

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useBlockProps } from '@wordpress/block-editor';
+import { Icon, info } from '@wordpress/icons';
+import ProductControl from '@woocommerce/editor-components/product-control';
+import { SelectedOption } from '@woocommerce/block-hocs';
+import {
+	Placeholder,
+	// @ts-expect-error Using experimental features
+	__experimentalHStack as HStack,
+	// @ts-expect-error Using experimental features
+	__experimentalText as Text,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import type { ProductCollectionEditComponentProps } from '../types';
+import { getCollectionByName } from '../collections';
+
+const ProductPicker = ( props: ProductCollectionEditComponentProps ) => {
+	const blockProps = useBlockProps();
+	const attributes = props.attributes;
+
+	const collection = getCollectionByName( attributes.collection );
+	if ( ! collection ) {
+		return;
+	}
+
+	return (
+		<div { ...blockProps }>
+			<Placeholder className="wc-blocks-product-collection__editor-product-picker">
+				<HStack alignment="center">
+					<Icon
+						icon={ info }
+						className="wc-blocks-product-collection__info-icon"
+					/>
+					<Text>
+						<strong>{ collection.title }</strong> requires a product
+						to be selected in order to display associated items.
+					</Text>
+				</HStack>
+				<ProductControl
```

Comment:
After testing this PR I found myself wondering whether or not this picker should include the `showVariations={ true }` option. A variation is an "imaginary" product of sorts and might not make sense to be selectable. On the other hand though, individual variations can have their own prices, code hook integration, and more. It seems reasonable perhaps that we can start here by doing absolutely nothing :smile:

Reply:
Good point! We actually discussed this in slack (p1723038190991919-slack-C05UMTA7W0P). We decided to not show variations in product picker 🙂 Glad to know that you are having same thoughts while reviewing the PR 😀

Reply:
Oh! Variations can't have upsells and cross-sells, fascinating! That _definitely_ means we should hide them :smile: